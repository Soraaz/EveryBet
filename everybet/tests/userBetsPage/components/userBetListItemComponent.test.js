import React from "react";
import {Platform, TouchableNativeFeedback, TouchableOpacity} from "react-native";
import renderer from "react-test-renderer";
import {shallow} from "enzyme";

import UserBetListItemComponent from "../../../src/userBetsPage/components/userBetListItemComponent";

describe("UserBetListItemComponent", () => {
	let bet = {
		"id": 1,
		"name": "TestBet",
		"description": "TestBetDesc",
		"validate": 1,
		"finished": 0,
		"deadline": 1111111,
		"correctAnswerId": 1,
		"coinsRedistributed": 10,
		"answers": [],
		"categories": [],
		"open": false
	};

	let categories = [
		{
			"id": 1,
			"name": "Art",
			"icon": "art.png"
		},
		{
			"id": 2,
			"name": "Finance",
			"icon": "finance.png"
		},
		{
			"id": 11,
			"name": "Sport",
			"icon": "sport.png"
		}];

	describe("rendering", () => {
		it("Should not render a bet if the filter doesn't exist", () => {
			const tree = renderer
				.create(<UserBetListItemComponent
					onPress={() => {}}
					bet={bet}
					categoryList={categories}
					filter={"test"}
					type={"progress"}
				/>)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});

		describe("android", () => {
			beforeEach(() => {
				Platform.OS = "android";
				bet.categories = [];
				bet.open = false;
			});

			it("Should render a bet in progress with the description closed and no category icon on android", () => {
				const tree = renderer
					.create(<UserBetListItemComponent
						onPress={() => {}}
						bet={bet}
						categoryList={categories}
						filter={"Toutes"}
						type={"progress"}
					/>)
					.toJSON();
				expect(tree).to.matchSnapshot();
			});

			it("Should render a bet in progress with the description closed and its category icon on android", () => {
				bet.categories = [{
					"id": 11,
					"name": "Sport",
					"icon": "sport.png"
				}];
				const tree = renderer
					.create(<UserBetListItemComponent
						onPress={() => {}}
						bet={bet}
						categoryList={categories}
						filter={"Toutes"}
						type={"progress"}
					/>)
					.toJSON();
				expect(tree).to.matchSnapshot();
			});

			it("Should render a bet in progress with the description opened and no category icon on android", () => {
				bet.open = true;
				const tree = renderer
					.create(<UserBetListItemComponent
						onPress={() => {}}
						bet={bet}
						categoryList={categories}
						filter={"Toutes"}
						type={"progress"}
					/>)
					.toJSON();
				expect(tree).to.matchSnapshot();
			});

			it("Should render a bet in progress with the description opened and its category icon on android", () => {
				bet.categories = [{
					"id": 11,
					"name": "Sport",
					"icon": "sport.png"
				}];
				bet.open = true;
				const tree = renderer
					.create(<UserBetListItemComponent
						onPress={() => {}}
						bet={bet}
						categoryList={categories}
						filter={"Toutes"}
						type={"progress"}
					/>)
					.toJSON();
				expect(tree).to.matchSnapshot();
			});

			it("Should render a bet which has ended with the description closed and no category icon on android", () => {
				const tree = renderer
					.create(<UserBetListItemComponent
						onPress={() => {}}
						bet={bet}
						categoryList={categories}
						filter={"Toutes"}
						type={"historic"}
					/>)
					.toJSON();
				expect(tree).to.matchSnapshot();
			});

			it("Should render a bet which has ended with the description closed and its category icon on android", () => {
				bet.categories = [{
					"id": 11,
					"name": "Sport",
					"icon": "sport.png"
				}];
				const tree = renderer
					.create(<UserBetListItemComponent
						onPress={() => {}}
						bet={bet}
						categoryList={categories}
						filter={"Toutes"}
						type={"historic"}
					/>)
					.toJSON();
				expect(tree).to.matchSnapshot();
			});

			it("Should render a bet which has ended with the description opened and no category icon on android", () => {
				bet.open = true;
				const tree = renderer
					.create(<UserBetListItemComponent
						onPress={() => {}}
						bet={bet}
						categoryList={categories}
						filter={"Toutes"}
						type={"historic"}
					/>)
					.toJSON();
				expect(tree).to.matchSnapshot();
			});

			it("Should render a bet which has ended with the description opened and its category icon on android", () => {
				bet.categories = [{
					"id": 11,
					"name": "Sport",
					"icon": "sport.png"
				}];
				bet.open = true;
				const tree = renderer
					.create(<UserBetListItemComponent
						onPress={() => {}}
						bet={bet}
						categoryList={categories}
						filter={"Toutes"}
						type={"historic"}
					/>)
					.toJSON();
				expect(tree).to.matchSnapshot();
			});
		});

		describe("ios", () => {
			beforeEach(() => {
				Platform.OS = "ios";
				bet.categories = [];
				bet.open = false;
			});

			it("Should render a bet in progress with the description closed and no category icon on ios", () => {
				const tree = renderer
					.create(<UserBetListItemComponent
						onPress={() => {}}
						bet={bet}
						categoryList={categories}
						filter={"Toutes"}
						type={"progress"}
					/>)
					.toJSON();
				expect(tree).to.matchSnapshot();
			});

			it("Should render a bet in progress with the description closed and its category icon on ios", () => {
				bet.categories = [{
					"id": 11,
					"name": "Sport",
					"icon": "sport.png"
				}];
				const tree = renderer
					.create(<UserBetListItemComponent
						onPress={() => {}}
						bet={bet}
						categoryList={categories}
						filter={"Toutes"}
						type={"progress"}
					/>)
					.toJSON();
				expect(tree).to.matchSnapshot();
			});

			it("Should render a bet in progress with the description opened and no category icon on ios", () => {
				bet.open = true;
				const tree = renderer
					.create(<UserBetListItemComponent
						onPress={() => {}}
						bet={bet}
						categoryList={categories}
						filter={"Toutes"}
						type={"progress"}
					/>)
					.toJSON();
				expect(tree).to.matchSnapshot();
			});

			it("Should render a bet in progress with the description opened and its category icon on ios", () => {
				bet.categories = [{
					"id": 11,
					"name": "Sport",
					"icon": "sport.png"
				}];
				bet.open = true;
				const tree = renderer
					.create(<UserBetListItemComponent
						onPress={() => {}}
						bet={bet}
						categoryList={categories}
						filter={"Toutes"}
						type={"progress"}
					/>)
					.toJSON();
				expect(tree).to.matchSnapshot();
			});

			it("Should render a bet which has ended with the description closed and no category icon on ios", () => {
				const tree = renderer
					.create(<UserBetListItemComponent
						onPress={() => {}}
						bet={bet}
						categoryList={categories}
						filter={"Toutes"}
						type={"historic"}
					/>)
					.toJSON();
				expect(tree).to.matchSnapshot();
			});

			it("Should render a bet which has ended with the description closed and its category icon on ios", () => {
				bet.categories = [{
					"id": 11,
					"name": "Sport",
					"icon": "sport.png"
				}];
				const tree = renderer
					.create(<UserBetListItemComponent
						onPress={() => {}}
						bet={bet}
						categoryList={categories}
						filter={"Toutes"}
						type={"historic"}
					/>)
					.toJSON();
				expect(tree).to.matchSnapshot();
			});

			it("Should render a bet which has ended with the description opened and no category icon on ios", () => {
				bet.open = true;
				const tree = renderer
					.create(<UserBetListItemComponent
						onPress={() => {}}
						bet={bet}
						categoryList={categories}
						filter={"Toutes"}
						type={"historic"}
					/>)
					.toJSON();
				expect(tree).to.matchSnapshot();
			});

			it("Should render a bet which has ended with the description opened and its category icon on ios", () => {
				bet.categories = [{
					"id": 11,
					"name": "Sport",
					"icon": "sport.png"
				}];
				bet.open = true;
				const tree = renderer
					.create(<UserBetListItemComponent
						onPress={() => {}}
						bet={bet}
						categoryList={categories}
						filter={"Toutes"}
						type={"historic"}
					/>)
					.toJSON();
				expect(tree).to.matchSnapshot();
			});
		});
	});

	describe("functionalities", () => {
		let spy;
		let wrapper;

		beforeEach(() => {
			spy = sinon.spy();
			wrapper = shallow(<UserBetListItemComponent
				onPress={spy}
				bet={bet}
				categoryList={categories}
				filter={"Toutes"}
				type={"progress"}
			/>);
		});

		describe("android", () => {
			beforeAll(() => {
				Platform.OS = "android";
			});

			it("Should call the callback if bet is pressed on android", () => {
				wrapper.find(TouchableNativeFeedback).first().props().onPress();
				expect(spy).to.be.calledOnce;
			});
		});

		describe("ios", () => {
			beforeAll(() => {
				Platform.OS = "ios";
			});

			it("Should call the callback if bet is pressed on ios", () => {
				Platform.OS = "ios";
				wrapper.find(TouchableOpacity).first().props().onPress();
				expect(spy).to.be.calledOnce;
			});
		});
	});
});
