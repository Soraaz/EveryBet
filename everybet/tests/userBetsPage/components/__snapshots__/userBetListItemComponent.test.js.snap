// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBetListItemComponent rendering Should not render a bet if the filter doesn't exist 1`] = `null`;

exports[`UserBetListItemComponent rendering android Should render a bet in progress with the description closed and its category icon on android 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffbcbc",
          "borderColor": "red",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "center",
          "margin": 10,
          "width": 300,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#333333",
          "margin": 20,
        }
      }
    >
      TouchableNativeFeedback is not supported on this platform!
    </Text>
  </View>
</View>
`;

exports[`UserBetListItemComponent rendering android Should render a bet in progress with the description closed and no category icon on android 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffbcbc",
          "borderColor": "red",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "center",
          "margin": 10,
          "width": 300,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#333333",
          "margin": 20,
        }
      }
    >
      TouchableNativeFeedback is not supported on this platform!
    </Text>
  </View>
</View>
`;

exports[`UserBetListItemComponent rendering android Should render a bet in progress with the description opened and its category icon on android 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffbcbc",
          "borderColor": "red",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "center",
          "margin": 10,
          "width": 300,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#333333",
          "margin": 20,
        }
      }
    >
      TouchableNativeFeedback is not supported on this platform!
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFFBF",
        "borderBottomLeftRadius": 5,
        "borderBottomRightRadius": 5,
        "flex": 1,
        "justifyContent": "flex-start",
        "marginHorizontal": 40,
        "paddingBottom": 20,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 15,
          "marginHorizontal": 20,
        }
      }
    >
      TestBetDesc
    </Text>
    <Text
      style={
        Object {
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 15,
          "marginHorizontal": 20,
        }
      }
    >
      Votre réponse est : undefined
    </Text>
  </View>
</View>
`;

exports[`UserBetListItemComponent rendering android Should render a bet in progress with the description opened and no category icon on android 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffbcbc",
          "borderColor": "red",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "center",
          "margin": 10,
          "width": 300,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#333333",
          "margin": 20,
        }
      }
    >
      TouchableNativeFeedback is not supported on this platform!
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFFBF",
        "borderBottomLeftRadius": 5,
        "borderBottomRightRadius": 5,
        "flex": 1,
        "justifyContent": "flex-start",
        "marginHorizontal": 40,
        "paddingBottom": 20,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 15,
          "marginHorizontal": 20,
        }
      }
    >
      TestBetDesc
    </Text>
    <Text
      style={
        Object {
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 15,
          "marginHorizontal": 20,
        }
      }
    >
      Votre réponse est : undefined
    </Text>
  </View>
</View>
`;

exports[`UserBetListItemComponent rendering android Should render a bet which has ended with the description closed and its category icon on android 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffbcbc",
          "borderColor": "red",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "center",
          "margin": 10,
          "width": 300,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#333333",
          "margin": 20,
        }
      }
    >
      TouchableNativeFeedback is not supported on this platform!
    </Text>
  </View>
</View>
`;

exports[`UserBetListItemComponent rendering android Should render a bet which has ended with the description closed and no category icon on android 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffbcbc",
          "borderColor": "red",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "center",
          "margin": 10,
          "width": 300,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#333333",
          "margin": 20,
        }
      }
    >
      TouchableNativeFeedback is not supported on this platform!
    </Text>
  </View>
</View>
`;

exports[`UserBetListItemComponent rendering android Should render a bet which has ended with the description opened and its category icon on android 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffbcbc",
          "borderColor": "red",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "center",
          "margin": 10,
          "width": 300,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#333333",
          "margin": 20,
        }
      }
    >
      TouchableNativeFeedback is not supported on this platform!
    </Text>
  </View>
</View>
`;

exports[`UserBetListItemComponent rendering android Should render a bet which has ended with the description opened and no category icon on android 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffbcbc",
          "borderColor": "red",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "center",
          "margin": 10,
          "width": 300,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#333333",
          "margin": 20,
        }
      }
    >
      TouchableNativeFeedback is not supported on this platform!
    </Text>
  </View>
</View>
`;

exports[`UserBetListItemComponent rendering ios Should render a bet in progress with the description closed and its category icon on ios 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF7F",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginHorizontal": 35,
          "marginTop": 25,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "[object Object]images/categories/sport.png",
          }
        }
        style={
          Object {
            "height": 40,
            "marginHorizontal": 10,
            "marginVertical": 10,
            "width": 40,
          }
        }
      />
      <Text
        style={
          Object {
            "flex": 1,
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        TestBet
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/bet_closed.png",
          }
        }
        style={
          Object {
            "height": 10,
            "marginHorizontal": 10,
            "width": 17,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`UserBetListItemComponent rendering ios Should render a bet in progress with the description closed and no category icon on ios 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF7F",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginHorizontal": 35,
          "marginTop": 25,
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        TestBet
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/bet_closed.png",
          }
        }
        style={
          Object {
            "height": 10,
            "marginHorizontal": 10,
            "width": 17,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`UserBetListItemComponent rendering ios Should render a bet in progress with the description opened and its category icon on ios 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF7F",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginHorizontal": 35,
          "marginTop": 25,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "[object Object]images/categories/sport.png",
          }
        }
        style={
          Object {
            "height": 40,
            "marginHorizontal": 10,
            "marginVertical": 10,
            "width": 40,
          }
        }
      />
      <Text
        style={
          Object {
            "flex": 1,
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        TestBet
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/bet_opened.png",
          }
        }
        style={
          Object {
            "height": 10,
            "marginHorizontal": 10,
            "width": 17,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFFBF",
        "borderBottomLeftRadius": 5,
        "borderBottomRightRadius": 5,
        "flex": 1,
        "justifyContent": "flex-start",
        "marginHorizontal": 40,
        "paddingBottom": 20,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 15,
          "marginHorizontal": 20,
        }
      }
    >
      TestBetDesc
    </Text>
    <Text
      style={
        Object {
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 15,
          "marginHorizontal": 20,
        }
      }
    >
      Votre réponse est : undefined
    </Text>
  </View>
</View>
`;

exports[`UserBetListItemComponent rendering ios Should render a bet in progress with the description opened and no category icon on ios 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF7F",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginHorizontal": 35,
          "marginTop": 25,
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        TestBet
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/bet_opened.png",
          }
        }
        style={
          Object {
            "height": 10,
            "marginHorizontal": 10,
            "width": 17,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFFBF",
        "borderBottomLeftRadius": 5,
        "borderBottomRightRadius": 5,
        "flex": 1,
        "justifyContent": "flex-start",
        "marginHorizontal": 40,
        "paddingBottom": 20,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 15,
          "marginHorizontal": 20,
        }
      }
    >
      TestBetDesc
    </Text>
    <Text
      style={
        Object {
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 15,
          "marginHorizontal": 20,
        }
      }
    >
      Votre réponse est : undefined
    </Text>
  </View>
</View>
`;

exports[`UserBetListItemComponent rendering ios Should render a bet which has ended with the description closed and its category icon on ios 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF7F",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginHorizontal": 35,
          "marginTop": 25,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "[object Object]images/categories/sport.png",
          }
        }
        style={
          Object {
            "height": 40,
            "marginHorizontal": 10,
            "marginVertical": 10,
            "width": 40,
          }
        }
      />
      <Text
        style={
          Object {
            "flex": 1,
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        TestBet
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/bet_closed.png",
          }
        }
        style={
          Object {
            "height": 10,
            "marginHorizontal": 10,
            "width": 17,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`UserBetListItemComponent rendering ios Should render a bet which has ended with the description closed and no category icon on ios 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF7F",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginHorizontal": 35,
          "marginTop": 25,
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        TestBet
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/bet_closed.png",
          }
        }
        style={
          Object {
            "height": 10,
            "marginHorizontal": 10,
            "width": 17,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`UserBetListItemComponent rendering ios Should render a bet which has ended with the description opened and its category icon on ios 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF7F",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginHorizontal": 35,
          "marginTop": 25,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "[object Object]images/categories/sport.png",
          }
        }
        style={
          Object {
            "height": 40,
            "marginHorizontal": 10,
            "marginVertical": 10,
            "width": 40,
          }
        }
      />
      <Text
        style={
          Object {
            "flex": 1,
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        TestBet
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/bet_opened.png",
          }
        }
        style={
          Object {
            "height": 10,
            "marginHorizontal": 10,
            "width": 17,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`UserBetListItemComponent rendering ios Should render a bet which has ended with the description opened and no category icon on ios 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF7F",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginHorizontal": 35,
          "marginTop": 25,
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        TestBet
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/bet_opened.png",
          }
        }
        style={
          Object {
            "height": 10,
            "marginHorizontal": 10,
            "width": 17,
          }
        }
      />
    </View>
  </View>
</View>
`;
